TelepresenceBot/mobile/src/main/java/com/novoda/tpbot/bot/BotActivity.java

```diff
@@ -35,6 +50,68 @@ protected void onCreate(Bundle savedInstanceState) {
 
         RecyclerView.Adapter adapter = new DirectionAdapter(LayoutInflater.from(this), Collections.<Direction>emptyList());
         directions.setAdapter(adapter);
+
+        ControllerView controllerView = Views.findById(this, R.id.bot_debug_controls);
+        controllerView.setControllerListener(controllerListener);
+
+        handler = new Handler();
+    }
+
+    private final ControllerListener controllerListener = new ControllerListener() {
+
+        @Override
+        public void onDirectionPressed(Direction direction) {
+            startRepeatingCommand(direction.rawCommand());
+        }
+
+        @Override
+        public void onDirectionReleased(Direction direction) {
+            stopRepeatingCommand(direction.rawCommand());
+        }
+
+        @Override
+        public void onLazersFired() {
+            // no-op for debug
+        }
+
+        @Override
+        public void onLazersReleased() {
+            // no-op for debug
+        }
+    };
+
+    private void startRepeatingCommand(String command) {
+        currentCommand = command;
+        sendCommand(currentCommand);
+        handler.postDelayed(repeatCommand, COMMAND_REPEAT_DELAY);
+    }
+
+    private void stopRepeatingCommand(String command) {
+        if (currentCommand != null && currentCommand.equals(command)) {
+            handler.removeCallbacks(repeatCommand);
+            currentCommand = null;
+        }
+    }
+
+    private Runnable repeatCommand = new Runnable() {
+        @Override
+        public void run() {
+            sendCommand(BotActivity.this.currentCommand);
```

Comment:
These 2 lines can maybe extracted into a method named `sendRepeatingCommand(String)`
What do you think?